### PR TITLE
C23_WMDE_Desktop_DE_18

### PR DESCRIPTION
**Pull request by Sperling-0**
Based on the VAR of test 15.
The variant uses a different copy.

Ticket: https://phabricator.wikimedia.org/T352513

### CODE REVIEW

File: banners/desktop/banner_var.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { createVueApp } from '@src/createVueApp';
 import './styles/styles_var.scss';
 
 import BannerConductor from '@src/components/BannerConductor/FallbackBannerConductor.vue';
-import Banner from './components/BannerVar.vue';
+import Banner from './components/BannerCtrl.vue';
```
Comment (Abban) [inline banners/desktop/banner_var.ts]: This should import BannerVar

File: banners/desktop/banner_var.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { createVueApp } from '@src/createVueApp';
 import './styles/styles_var.scss';
```
Comment (Abban) [inline banners/desktop/banner_var.ts]: This should import `styles.scss`

**Review by Abban — CHANGES_REQUESTED** [review]
A couple of things needed:

- The imports in the VAR banner entry point are wrong, and it currently has the same content as the CTRL
- There's a few unused files left in, you should delete:
	- `components/MainBannerVar.vue`
	- `styles/styles_var`
	- `messages_var.ts`
	- `visitorsVsDailyDonorsSentence.ts`

File: banners/desktop/banner_var.ts (hunk 1 above)
Comment (Sperling-0) [reply banners/desktop/banner_var.ts]: Done

File: banners/desktop/banner_var.ts (hunk 2 above)
Comment (Sperling-0) [reply banners/desktop/banner_var.ts]: Done

**Review by gbirke — APPROVED** [review]

File: banners/desktop/banner_var.ts [hunk 3]
```diff
@@ -75,6 +74,5 @@ app.provide( 'formItems', createFormItems( translator, currencyFormatter.euroAmo
 app.provide( 'formActions', createFormActions( page.getTracking(), impressionCount, { des: '1' } ) );
 app.provide( 'tracker', tracker );
 app.provide( 'dailyDonorAverage', dailyDonorAverage );
```
Comment (Abban) [inline banners/desktop/banner_var.ts]: This is no longer being used, you can delete it. You can also delete the `dailyDonorAverage` const and the `LinearDailyDonorAverage` and `IntegerDe` imports.

**Review by Abban — APPROVED** [review]
Looks good, just a little clean up left.

When you add the description to the git commits you should add an extra blank line between the title and the description. Without the extra break it outputs the first 2 lines in the IDE and git log.

**Comment by Abban** [comment]
Can you add tests for the already donated modal into the banner tests:

```js
describe( 'Already Donated', () => {
	test.each( [
		[ 'expectShowsAlreadyDonatedModal' ],
		[ 'expectHidesAlreadyDonatedModal' ],
		[ 'expectFiresMaybeLaterEvent' ],
		[ 'expectFiresGoAwayEvent' ]
	] )( '%s', async ( testName: string ) => {
		await alreadyDonatedModalFeatures[ testName ]( getWrapper() );
	} );
} );
```

You'll need to change the `expectFiresMaybeLaterEvent` feature test to check for a `Close`rather than `MaybeLater`. You can get an example in: https://github.com/wmde/fundraising-banners/pull/301

File: banners/desktop/banner_var.ts (hunk 3 above)
Comment (Sperling-0) [reply banners/desktop/banner_var.ts]: Done.

**Comment by gbirke** [comment]
@Sperling-0 I have merged Desktop-17. You can rebase on main, always accepting your changes and throwing away the ones in `main`.

**Pull request merged**